CMSgov/onemac
# Oy2 22088 - remove title from top of details page

Story: https://qmacbis.atlassian.net/browse/OY2-22088
Endpoint: See github-actions bot comment

### Details

Remove Title as an element on top of details page

### Changes

- removed the title section from DetailSection.tsx
- removed usage of OneMACDetail.defaultTitle since it wasnt used

### Test Plan

1. Login as any user
2. Navigate to details page of any app k
3. Verify Title no longer appears at the top above the info cards
